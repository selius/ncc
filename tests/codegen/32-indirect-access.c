int main()
{
	struct SomeStruct {
		int a;
		float b;
	};

	struct ComplexStruct {
		int a;
		struct {
			int a;
		} b;
		float c;
	};

	struct SomeStruct s;
	struct ComplexStruct c;

	struct SomeStruct *ps;
	struct ComplexStruct *pc;

	ps = &s;
	pc = &c;

	ps->a = 5;
	ps->b = 3.78;

	__print_int(s.a);
	__print_float(s.b);

	pc->b.a = 99;

	__print_int(c.b.a);

	pc->a = pc->b.a;
	__print_int(c.a);

	__print_int(ps->a);
	__print_float(ps->b);
	__print_int(pc->b.a);
	__print_int(pc->a);
	
	return 0;
}
