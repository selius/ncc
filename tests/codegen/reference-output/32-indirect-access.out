5
3.780000
99
99
5
3.780000
99
99
